scripts/performance_setup.py

```diff
@@ -0,0 +1,362 @@
+        using_mono = True
+        args.configurations["LLVM"] = str(args.llvm)
+        args.configurations["MonoInterpreter"] = str(args.mono_interpreter)
+        args.configurations["MonoAOT"] = str(args.mono_aot)
+
+        # TODO: Validate if this exclusion filter is still needed
+        extra_bdn_arguments += ["--exclusion-filter", "*Perf_Image*", "*Perf_NamedPipeStream*"]
+        category_exclusions += ["NoMono"]
+
+        if args.mono_interpreter:
+            category_exclusions += ["NoInterpreter"]
+
+        if args.mono_aot:
+            category_exclusions += ["NoAOT"]
+
+    using_wasm = False
+    if args.wasm_bundle_directory is not None:
+        using_wasm = True
+        args.configurations["CompilationMode"] = "wasm"
+        if args.wasm_aot:
+            args.configurations["AOT"] = "true"
+            build_config = f"wasmaot.{build_config}"
+        else:
+            build_config = f"wasm.{build_config}"
+
+        if args.javascript_engine == "javascriptcore":
+            args.configurations["JSEngine"] = "javascriptcore"
+
+        category_exclusions += ["NoInterpreter", "NoWASM", "NoMono"]
+
+    if args.no_dynamic_pgo:
+        args.configurations["PGOType"] = "nodynamicpgo"
+
+    if args.physical_promotion:
+        args.configurations["PhysicalPromotionType"] = "physicalpromotion"
+
+    runtime_type = ""
+
+    if args.ios_mono:
+        runtime_type = "Mono"
+        args.configurations["iOSLlvmBuild"] = str(args.ios_llvm_build)
+        args.configurations["iOSStripSymbols"] = str(args.ios_strip_symbols)
+        args.configurations["RuntimeType"] = str(runtime_type)
+
+    if args.ios_nativeaot:
+        runtime_type = "NativeAOT"
+        args.configurations["iOSStripSymbols"] = str(args.ios_strip_symbols)
+        args.configurations["RuntimeType"] = str(runtime_type)
+
+    if category_exclusions:
+        extra_bdn_arguments += ["--category-exclusion-filter", *set(category_exclusions)]
+    
+    cleaned_branch_name = "main"
+    if args.branch is not None and args.branch.startswith("refs/heads/release"):
+        cleaned_branch_name = args.branch.replace("refs/heads/", "")
+
+    setup_arguments = CiSetupArgs(
+        channel=cleaned_branch_name,
+        queue=args.queue,
+        build_configs=[f"{k}={v}" for k, v in args.configurations.items()],
+        architecture=args.architecture,
+        get_perf_hash=True
+    )
+
+    if args.build_number is not None:
+        setup_arguments.build_number = args.build_number
+
+    if args.repository is not None:
+        setup_arguments.repository = f"https://github.com/{args.repository}"
+
+    if args.branch is not None:
+        setup_arguments.branch = args.branch
+
+    if args.commit_sha is not None:
+        setup_arguments.commit_sha = args.commit_sha
+
+    if not args.internal:
+        setup_arguments.not_in_lab = True
+
+    # TODO: Figure out if this should be the runtime or performance commit time, or if we need to capture both
```

Comment:
I think we should either be just the runtime commit time, or both the runtime and performance times. I prefer to have both, even if the performance repo commit time gets added as additional data.

Reply:
At the moment it's only practical to get the performance repo commit time, as only the core root folder is uploaded to the crank agent. To get the runtime commit information we may need to look at sending up the `.git` directory somehow to crank. I'll leave this TODO in here since I don't think I will be able to resolve it just yet.

Reply:
Sounds good to me.

Reply:
Sorry to unresolve, but I am confused here. Shouldn't we be getting the commit time from calling ci_setup.py? Specifically, we call `dotnet.get_commit_time` and this properly gets the commit based on what repo we are running against.

Reply:
Oh, I see I missed that we don't follow that code path here because we are running against something local. It does feel like this is something we could grab locally when we send the job off to crank, but that feels outside the scope for sure.